lambdas/pkgpush/index.py

```diff
@@ -172,6 +186,55 @@
 logger = get_quilt_logger()
 
 
+def calculate_pkg_hashes(s3_client, pkg):
+    entries = []
+    for lk, entry in pkg.walk():
+        if entry.hash is not None:
+            continue
+        if entry.size > S3_HASH_LAMBDA_MAX_FILE_SIZE_BYTES:
+            raise FileTooLargeForHashing(lk)
+
+        entries.append(entry)
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=S3_HASH_LAMBDA_CONCURRENCY) as pool:
```

Comment:
If I understand all this correctly, the only purpose of the thread pool is to wait for multiple lambas - and no actual work is being done there, right?

I see that the lamba API supports async calls using `InvocationType='Event'` - would that be useful? Though if the async API makes everything unnecessarily complicated, then thread pool is fine. (I can't tell how to wait for the results from the async call...)

Reply:
> I see that the lamba API supports async calls using `InvocationType='Event'` - would that be useful? Though if the async API makes everything unnecessarily complicated, then thread pool is fine. (I can't tell how to wait for the results from the async call...)

It's not Python async and I don't think it could be useful.